src/components/App.js

```diff
@@ -20,16 +106,293 @@ export default class App extends Component {
     'Samstag'
   ]
 
-  getDay() {
-    const dayIndex = new Date().getDay()
-    return this.daysArray[dayIndex]
-  }
-
   render() {
     return (
       <Wrapper>
-        <JournalCard day={this.getDay()} date={new Date()} />
+        <JournalCard
+          addingEntry={this.state.addingEntry}
+          editEntry={this.state.editEntry}
+          day={this.getDay()}
+          date={new Date()}
+          data={this.props.state}
+          handleSubmit={this.handleSubmit}
+          renderJournalTexts={this.renderJournalTexts}
+          renderAmount={this.renderAmount}
+          renderEnergy={this.renderEnergy}
+          renderMood={this.renderMood}
+          renderTasks={this.renderTasks}
+          toggleEntryWindow={this.toggleEntryWindow}
+          replaceEntry={() => this.replaceEntry(this.state.editIndex)}
+        />
       </Wrapper>
     )
   }
+
+  getDay() {
+    const dayIndex = new Date().getDay()
+    return this.daysArray[dayIndex]
+  }
+
+  renderAnyTags(type, typeName) {
+    return type.map(item => (
+      <EntryTag
+        text={item.text}
+        selected={item.selected}
+        onClick={() => this.selectClickedTag(item.id, type, typeName)}
+        key={item.id}
+      />
+    ))
+  }
+
+  renderAmount = () => {
+    const selectedTask = this.state.tasks.find(task => task.selected === true)
+    if (selectedTask == null) {
+      return null
+    } else {
+      if (selectedTask.amount === true)
+        return (
+          <React.Fragment>
+            <Separator text="Amount" />
+            {this.renderAnyTags(this.state.amount, 'amount')}
+          </React.Fragment>
+        )
+    }
+  }
+
+  renderEnergy = () => {
+    const selectedTask = this.state.tasks.find(task => task.selected === true)
+    if (selectedTask == null) {
+      return null
+    } else {
+      if (selectedTask.energy === true)
+        return (
+          <React.Fragment>
+            <Separator text="Energy" />
+            {this.renderAnyTags(this.state.energy, 'energy')}
+          </React.Fragment>
+        )
+    }
+  }
+
+  renderMood = () => {
+    const selectedTask = this.state.tasks.find(task => task.selected === true)
+    if (selectedTask == null) {
+      return null
+    } else {
+      if (selectedTask.mood === true)
+        return (
+          <React.Fragment>
+            <Separator text="Mood" />
+            {this.renderAnyTags(this.state.mood, 'mood')}
+          </React.Fragment>
+        )
+    }
+  }
+
+  renderTasks = () => {
+    return this.renderAnyTags(this.state.tasks, 'tasks')
+  }
+
+  handleSubmit = index => {
+    const task = this.state.tasks.find(item => item.selected === true)
+    const amount = this.state.amount.find(item => item.selected === true)
+    const energy = this.state.energy.find(item => item.selected === true)
+    const mood = this.state.mood.find(item => item.selected === true)
+
+    const entry = { task: task, amount: amount, energy: energy, mood: mood }
```

Comment:
I think there's a short notation for that case. Like: const entry = {task, amount, energy, mood}
Can only be used when key equals value